web3/eth.py

```diff
@@ -531,9 +531,16 @@ def wait_for_transaction_receipt(
         try:
             return wait_for_transaction_receipt(self.web3, transaction_hash, timeout, poll_latency)
         except Timeout:
+            transaction_hash_bytes = transaction_hash
+
+            # to_hex function does not accept string parameters
+            # Convert to bytes if needed
+            if type(transaction_hash) == str:
+                transaction_hash_bytes = transaction_hash.encode('utf-8')
+
             raise TimeExhausted(
                 "Transaction {} is not in the chain, after {} seconds".format(
-                    to_hex(transaction_hash),
+                    to_hex(transaction_hash_bytes),
```

Comment:
Instead of trying to do the `encode` on line 539, you can just do something like: 
```suggestion
                    HexBytes(transaction_hash),
```
And then I believe that line 542 will need to become: 
```
"Transaction {!r} is not in the chain, after {} seconds".format(
```